earthly/cspell/Earthfile

```diff
@@ -0,0 +1,38 @@
+# cspell UDCs and Containers.
+VERSION 0.7
+
+CSPELL:
+    # Spell checking all docs and code is done with cspell
+    # See: cspell.org
+    COMMAND
+
+    # Where we want to run the `lint` from.
+    ARG --required src
+
+    # Unlikely to need to change this.
+    ARG cfg_file=cspell.json
+
+    RUN --no-cache if [[ ! -f $src/$cfg_file ]]; then \
+        (>&2 echo "$src/$cfg_file does not exist. Can not check spelling."); \
+        exit 1; \
+    fi
+
+    RUN --no-cache echo Spell Checking Recursively from: $src
+
+    WITH DOCKER --pull ghcr.io/streetsidesoftware/cspell:latest
```

Comment:
I still think it's worth considering installing this in its native package form rather than using `WITH DOCKER`.

Reply:
I think that this is a good way to enable swapping the cspell image, which I assume is already optimized and loaded with the configured binary.

Reply:
I do not believe there is any justifiable return on investment in re-constructing this already working container.  Its out of scope for this PR.

Reply:
Am I misunderstanding what your saying here?
Its not clear if your suggesting we re-make the cspell image, or we do this in an alternative way?

Reply:
I'm just looking to standardize the way we include external tools. Historically, for something like `cspell`, I would have done:

```
FROM node:20-bookworm
RUN npm install -g git+https://github.com/streetsidesoftware/cspell-cli
RUN cspell lint .
```

I've typically avoided `WITH DOCKER` because of some prior experiences where it was:

1. Slow
2. Didn't seem to cache the way I would have expected (i.e. I'm not sure if DinD caches the image)
3. Involved more effort to figure out how to user the Docker image correctly rather than just run the binary itself

In this case, the second one doesn't apply because we're technically using the local workstation as a cache (with `LOCALLY`), but that also sort of breaks the "isolation" that Earthly is providing (i.e. we're now creating a side effect). 

Whatever we want to do, I just want to make sure we do it across all Earthfiles for the sake of consistency.

Reply:
@jmgilman 
Thank you for your insights. Maintaining multiple third-party packages like mdlint and cspell may be challenging, given our limited resources. Finding a balance between standardization and maintenance overhead is crucial to ensure we remain focused on our core project deliverables. Your suggestion for consistency is well-noted, and finding a simplified, low-maintenance approach to include necessary external tools is a priority.

Reply:
Sure, I also only want to maintain what we need to. Let's go with it and we can always come back to this in the future if we start running into issues.